torch/csrc/TypeInfo.cpp

```diff
@@ -150,11 +152,19 @@ static PyObject* THPFInfo_min(THPFInfo* self, void*) {
   END_HANDLE_TH_ERRORS
 }
 
+#define AT_DISPATCH_IINFO_TYPES(TYPE, NAME, ...) \
+  AT_DISPATCH_V2(                                \
+      TYPE, NAME, AT_WRAP(__VA_ARGS__), AT_EXPAND(AT_INTEGRAL_TYPES_V2))
+
 static PyObject* THPIInfo_max(THPIInfo* self, void*) {
   HANDLE_TH_ERRORS
   if (at::isIntegralType(self->type, /*includeBool=*/false)) {
-    return AT_DISPATCH_INTEGRAL_TYPES(self->type, "max", [] {
-      return THPUtils_packInt64(std::numeric_limits<scalar_t>::max());
+    return AT_DISPATCH_IINFO_TYPES(self->type, "max", [] {
+      if (std::is_unsigned<scalar_t>::value) {
```

Comment:
nit: here and below since we had quite a few lint PRs fixing these
```suggestion
      if (std::is_unsigned_v<scalar_t>) {
```